src/net/tcp/listener.rs

```diff
@@ -43,10 +46,19 @@ impl TcpListener {
     /// Binding with a port number of 0 will request that the OS assigns a port
     /// to this listener. The port allocated can be queried via the `local_addr`
     /// method.
-    pub fn bind(socket_addr: SocketAddr) -> io::Result<TcpListener> {
-        let socket = Socket::bind(socket_addr, libc::SOCK_STREAM)?;
-        socket.listen(1024)?;
-        Ok(TcpListener { inner: socket })
+    ///
+    /// [`ToSocketAddrs`]: trait@std::net::ToSocketAddrs
+    pub fn bind<A: ToSocketAddrs>(addr: A) -> io::Result<Self> {
```

Comment:
Let's keep this as a SocketAddr for now, to prevent blocking.

Reply:
I don't think bind is an operation on the critical path. Personally, I think use of the trait makes API more ergonomic.

Could you name a scenario where you are continually binding a new port frequently?

Reply:
Actually, after some reflection, I understand that this blocking call could block the whole thread. So I guess I am in agreement.